h2o-docs/src/product/explain.rst

```diff
@@ -0,0 +1,369 @@
+
+        # Reponse column
+        y = "quality"
+
+        # Split into train & test
+        splits = df.split_frame(ratios = [0.8], seed = 1)
+        train = splits[0]
+        test = splits[1]
+
+        # Run AutoML for 1 minute
+        aml = H2OAutoML(max_runtime_secs=60, seed=1)
+        aml.train(y=y, training_frame=train)
+
+        # Explain leader model & compare with all AutoML models 
+        exa = aml.explain(test)
+
+        # Explain a single H2O model (e.g. leader model from AutoML)
+        exm = aml.leader.explain(test)
+
+
+Notes:
+''''''
+
+In R, the ``H2OExplanation`` object will not be printed if you save it to an object.  If you save the object to a variable, you will be able to access the plots and associated metadata by inspecting the object.  Then you can ``print(exa)`` to print the explanation, or simply type ``exa`` and it will print as well.
+
+In Python, the ``H2OExplanation`` will always be printed, even if you save it to an object.  Once you save it to an object, however, if you want to print it again, you must do: ``from IPython.core.display import display`` and ``display(exa)``.
+
+
+
+Explain a single row prediction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``h2o.explain_row()`` function provides model explanations for a single row of test data. Using the previous code example, you can evaluate row-level behavior by specifying the ``row_index``:
+
+.. tabs::
+   .. code-tab:: r R
+
+        # Explain row 1 with all AutoML models
+        h2o.explain_row(aml, test, row_index = 1)
+
+        # Explain row 1 with a single model
+        h2o.explain_row(aml@leader, row_index = 1)
+
+   .. code-tab:: python
+
+        # Explain row 1 with all AutoML models
+        aml.explain_row(test, row_index=1)
+
+        # Explain row 1 with a single model
+        aml.leader.explain_row(test, row_index=1)
+
+
+Output: Explanations
+--------------------
+
+TO DO: Overview of the output object.  Add some plots
+
+The output from these functions is an ``H2OExplanation``, which is essentially a list of explanations.  Most explanations are plots with some associated metadata that can be extracted.  The plots are ggplot2 objects in R and matplotlib figures in Python.  You can customize these plot objects as desired, after they are generated.  Note: You can also customize the plot objects before they are generated by utilizing the ``plot_overrides`` argument.
+
+The ``H2OExplanation`` is a named list in R and an ordered dictionary in Python, so explanations can be accessed by their name (same names as in ``include_explanations`` and ``exlude_exlpanations``).
+
+
+When ``h2o.explain()`` is provided a list of models, the following explanations will be generated by default:
+
+- Leaderboard (compare all models)
+- Confusion Matrix for Leader Model (classification only)
+- Residual Analysis for Leader Model (regression only)
+- Variable Importance of Top Base (non-Stacked) Model 
+- Variable Importance Heatmap (compare all models)
```

Comment:
`varimp_heatmap` is unable to compare Stacked Ensembles since they don't have the `varimp`

Reply:
thx